app/bundles/AssetBundle/Model/AssetModel.php

```diff
@@ -233,7 +233,10 @@ public function trackDownload($asset, $request = null, $code = '200', $systemEnt
 
         $download->setCode($code);
         $download->setIpAddress($ipAddress);
-        $download->setReferer($request->server->get('HTTP_REFERER'));
+
+        if (isset($request->server)) {
```

Comment:
With this, may be better to check `if (null !== $request)` since CLI will result in `$request = null` rather than an object.